UnitsNet/BaseDimensions.cs

```diff
@@ -0,0 +1,169 @@
+        }
+
+        public override int GetHashCode()
+        {
+            int hash = 17;
+            hash = hash * 23 + Length;
+            hash = hash * 23 + Mass;
+            hash = hash * 23 + Time;
+            hash = hash * 23 + Current;
+            hash = hash * 23 + Temperature;
+            hash = hash * 23 + Amount;
+            hash = hash * 23 + LuminousIntensity;
+            return hash;
+        }
+
+        public BaseDimensions Multiply(BaseDimensions right)
+        {
+            return new BaseDimensions(
+                Length + right.Length,
+                Mass + right.Mass,
+                Time + right.Time,
+                Current + right.Current,
+                Temperature + right.Temperature,
+                Amount + right.Amount,
+                LuminousIntensity + right.LuminousIntensity);
+        }
+
+        public BaseDimensions Divide(BaseDimensions right)
+        {
+            return new BaseDimensions(
+                Length - right.Length,
+                Mass - right.Mass,
+                Time - right.Time,
+                Current - right.Current,
+                Temperature - right.Temperature,
+                Amount - right.Amount,
+                LuminousIntensity - right.LuminousIntensity);
+        }
+
+#if !WINDOWS_UWP
+        public static bool operator ==(BaseDimensions left, BaseDimensions right)
+        {
+            return left.Equals(right);
+        }
+    
+        public static bool operator !=(BaseDimensions left, BaseDimensions right)
+        {
+            return !( left == right );
+        }
+
+        public static BaseDimensions operator *(BaseDimensions left, BaseDimensions right)
+        {
+            return left.Multiply(right);
+        }
+
+        public static BaseDimensions operator /(BaseDimensions left, BaseDimensions right)
+        {
+            return left.Divide(right);
+        }
+#endif
+
+        public override string ToString()
+        {
+            var sb = new StringBuilder();
+
+            AppendDimensionString(sb, "Length", Length);
+            AppendDimensionString(sb, "Mass", Mass);
+            AppendDimensionString(sb, "Time", Time);
+            AppendDimensionString(sb, "Current", Current);
+            AppendDimensionString(sb, "Temperature", Temperature);
+            AppendDimensionString(sb, "Amount", Amount);
+            AppendDimensionString(sb, "LuminousIntensity", LuminousIntensity);
+
+            return sb.ToString();
+        }
+
+        private static void AppendDimensionString( StringBuilder sb, string name, int value )
+        {
+            var absoluteValue = Math.Abs( value );
+
+            if( absoluteValue > 0 )
+            {
+                sb.AppendFormat( "[{0}]", name );
+
+                if( absoluteValue > 1 )
+                    sb.AppendFormat( "^{0}", value );
+            }
+        }
+
+        public int Length { get { return _length; } }
```

Comment:
In C#7 (or was it C#6?) you no longer need a backing field. You can simply have a property public int Mass { get; } that is then assigned its value in the constructor.